packages/components/menu/src/MenuList/MenuList.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { CommonProps, PropsWithHTMLElement } from '@contentful/f36-core';
+import { useMenuContext } from '../MenuContext';
+import { Popover } from '@contentful/f36-popover';
+import { cx } from 'emotion';
+import { getMenuListStyles } from './MenuList.styles';
+
+interface MenuListInternalProps extends CommonProps {
+  children?: React.ReactNode;
+
+  /**
+   * The item will be sticked to the top of MenuList
+   */
+  headerItem?: React.ReactElement;
+
+  /**
+   * The item will be sticked to the bottom of MenuList
+   */
+  footerItem?: React.ReactElement;
+}
+
+export type MenuListProps = PropsWithHTMLElement<MenuListInternalProps, 'div'>;
+
+const _MenuList = (props: MenuListProps, ref: React.Ref<HTMLDivElement>) => {
+  const {
+    children,
+    testId = 'cf-ui-menu-list',
+    className,
+    headerItem,
+    footerItem,
+    ...otherProps
+  } = props;
+
+  const { getMenuListProps } = useMenuContext();
+
+  const styles = getMenuListStyles();
+
+  return (
+    <Popover.Content
+      {...getMenuListProps(otherProps, ref)}
+      data-test-id={testId}
+      role="menu"
+      className={cx(styles.container, className)}
+    >
+      {headerItem &&
+        React.cloneElement(headerItem, {
+          className: cx(styles.headerItem, headerItem.props.className),
+        })}
+      {children}
+      {footerItem &&
+        React.cloneElement(footerItem, {
+          className: cx(styles.footerItem, footerItem.props.className),
+        })}
```

Comment:
Shouldn't we transform headerItem and footerItem into Components? (e.g.: `Menu.ListHeader`)

Reply:
@bgutsol said the reasoning basically was that by doing them components we cannot control a place where they will be put. `ListHeader` should always be the first and `ListFooter` should always be the last.

Reply:
I think we could
I noticed the same thing in the workbench, but since we didn't want to change the Workbench I didn't do it

The strategy would be to check the children of MenuList, then you can pick `ListHeader` and `ListFooter` by checking the `displayName` for example and render them always on top and bottom

That way it would not matter if someone does 

```tsx
<MenutItem>
<MenutItem>
<MenutItem>
<MenutFooter>
<MenutHeader>
```

I wanted to do that for the Sidebars in workbench, because now you need to position them correctly and set `position` prop

but that's not necessary

Reply:
I'll check what you're proposing and see if I'll be able to implement that.

Reply:
but, honestly, we don't need to do it now

We can do it later if we want... if we don't want it we just don't do it

Reply:
If you want to check, the Workbench is doing that for the header

```jsx
const header: React.ReactNode[] = [];
  const content: React.ReactNode[] = [];

  React.Children.forEach(
    children,
    // eslint-disable-next-line
    (child: any) => {
      if (child.type?.name === 'WorkbenchHeader') {
        header.push(child);
      } else {
        content.push(child);
      }
    },
  );
```

In v3, we had that `any` as the type of the child and I tried to replace it with proper typing but I couldn't
I tried using ReactNode, Child, Element, etc... all of them had some issue

I'm still trying to figure out the best way in TS for this pattern

Reply:
@gui-santos pushed the change, as you suggested.